samples/core/iris/iris_pipeline.py

```diff
@@ -0,0 +1,157 @@
+# example code and metadata library is relative to a GCS path.
+_pipeline_root = os.path.join(
+    'gs://jxzheng-helloworld', 'tfx_iris', kfp.dsl.RUN_ID_PLACEHOLDER
+)
+
+
+def _create_pipeline(
+    pipeline_name: Text, pipeline_root: Text
+) -> pipeline.Pipeline:
+  """Implements the Iris flowers pipeline with TFX."""
+  examples = external_input(_data_root_param)
+
+  # Brings data into the pipeline or otherwise joins/converts training data.
+  example_gen = CsvExampleGen(input=examples)
+
+  # Computes statistics over data for visualization and example validation.
+  statistics_gen = StatisticsGen(examples=example_gen.outputs['examples'])
+
+  # Generates schema based on statistics files.
+  infer_schema = SchemaGen(
+      statistics=statistics_gen.outputs['statistics'], infer_feature_shape=True
+  )
+
+  # Performs anomaly detection based on statistics and data schema.
+  validate_stats = ExampleValidator(
+      statistics=statistics_gen.outputs['statistics'],
+      schema=infer_schema.outputs['schema']
+  )
+
+  # Performs transformations and feature engineering in training and serving.
+  transform = Transform(
+      examples=example_gen.outputs['examples'],
+      schema=infer_schema.outputs['schema'],
+      module_file=_module_file_param
+  )
+
+  # Uses user-provided Python function that trains a model using TF-Learn.
+  trainer = Trainer(
+      module_file=_module_file_param,
+      custom_executor_spec=executor_spec.ExecutorClassSpec(GenericExecutor),
+      examples=transform.outputs['transformed_examples'],
+      transform_graph=transform.outputs['transform_graph'],
+      schema=infer_schema.outputs['schema'],
+      train_args=trainer_pb2.TrainArgs(num_steps=100),
+      eval_args=trainer_pb2.EvalArgs(num_steps=50)
+  )
+
+  # Uses TFMA to compute a evaluation statistics over features of a model.
+  model_analyzer = Evaluator(
+      examples=example_gen.outputs['examples'],
+      model=trainer.outputs['model'],
+      eval_config=tfma.EvalConfig(
+          model_specs=[tfma.ModelSpec(label_key='variety')],
+          slicing_specs=[tfma.SlicingSpec(feature_keys=['sepal_length'])]
+      )
+  )
+
+  return pipeline.Pipeline(
+      pipeline_name=pipeline_name,
+      pipeline_root=pipeline_root,
+      components=[
+          example_gen,
+          statistics_gen,
+          infer_schema,
+          validate_stats,
+          transform,
+          trainer,
+          model_analyzer,
+      ],
+      enable_cache=True,
+  )
+
+
+if __name__ == '__main__':
+  absl.logging.set_verbosity(absl.logging.INFO)
+  # Make sure the version of TFX image used is consistent with the version of
+  # TFX SDK. Here we use tfx:0.15.0 image.
+  config = kubeflow_dag_runner.KubeflowDagRunnerConfig(
+      kubeflow_metadata_config=kubeflow_dag_runner.
+      get_default_kubeflow_metadata_config(),
+      pipeline_operator_funcs=list(
+          filter(
+              lambda operator: operator.__name__.find('gcp_secret') == -1,
```

Comment:
Have we removed auto-applying `use_gcp_secret` in TFX master?

Reply:
We did in 0.21.0 formal release

I'm doing version bumping in another PR.

Reply:
Done